### DISCUSSION TITLE
Consider changes to slinger

### DISCUSSION THREAD

**Discussion by mdehrlich** [GerryDazoo/Slinger#446] [Ideas]
Gerry,  If you do another version, please consider changing the format string  %,.  With multiple configs, it was difficult to make a HTML remote using %s for box name, and for the status line.  I had to rewrite your code to use string replacement method.  Also pretty weird you have to double all the % signs in the style section for non-html remotes.  I would love to be able to use your releases without having to modify them. Thx

**Comment by mdehrlich** [comment]
Note that I have a nice FIOS HTML remote that works with 2 slingboxes + 3 resolutions each.  It automatically resizes larger on cellphones so I don't need to zoom to press buttons.  Also shows status with box name to help verify I have to correct remote for the config I am watching. As noted above, it took a few mods to the server code.

**Reply by GerryDazoo** [reply]
What mods did you have to make. I'll have a look and see what I can do. I hadn't considered adding box name to the status string all my remotes have a 
`<h1>BoxName</h1>`
 in the first line.

**Comment by mdehrlich** [comment]
Here are some screenshots.  The top left is pull down menu that lets me switch the remote to any box/config.  Code of HTML is also shown 
<img width="960" alt="2022-12-22 (1)" src="https://user-images.githubusercontent.com/8422167/209267276-d491741e-5069-46fe-ac02-3b0769c1f929.png">
<img width="960" alt="2022-12-22" src="https://user-images.githubusercontent.com/8422167/209267304-fb0143e9-7a1d-422d-896b-ec436eb4f0bd.png">

**Reply by GerryDazoo** [reply]
Very nice. But I was referring to mods of the server code.

**Comment by mdehrlich** [comment]
Here are the changes:
at line 1388 to insert the name into the html action= parameter:
if fn.lower().endswith('.html') :
        uri = '/' + section_name
        page = data.replace('Remote%s' , 'Remote' + uri, 1)
        cmds = parse_html(page)

at line 1472 to add name into status because why not?
 stati[box] =  box + ' Waiting for first update...'

at line 1575 to update status
            Page = Page.replace('Status:%s', 'Status:' + stati[streamer])

**Reply by mdehrlich** [reply]
I just figured out how to set the action attribute from the JS code.   We don't need the first replacement.

**Reply by mdehrlich** [reply]
I forgot about this one at line 1486
if streamer :
                    page = remotes[streamer][0]
                    page = page.replace('Status:%s', 'Status:' + stati[streamer])
                    return render_template_string( page )
